SSA/Projects/InstCombine/LLVM/Parser.lean

```diff
@@ -0,0 +1,15 @@
+import SSA.Core.MLIRSyntax.Parser
+import SSA.Projects.InstCombine.LLVM.EDSL
+import Init.Data.Repr
+
+open MLIR AST InstCombine
+def regionTransform (region : Region 0) : Except ParseError
+  (Σ (Γ' : Context) (eff : EffectKind) (ty : Ty), Com LLVM Γ' eff ty) :=
+    let res := mkCom (d:= LLVM) region
+    match res with
+      | Except.error e => Except.error s!"Error:\n{reprStr e}"
+      | Except.ok res => Except.ok res
+
+def parseComFromFile (fileName : String) :
+ IO (Option (Σ (Γ' : Context) (eff : EffectKind) (ty : InstCombine.Ty), Com LLVM Γ' eff ty)) := do
+ parseRegionFromFile fileName regionTransform
```

Comment:
I have to push back here @bollu. This is conceptually a library, that's why I split it here. The fact that it is *currently* only used by `opt` is just because we haven't implemented anything else, but this could be reused by anything that would want to parse from file.